EIPS/eip-7685.md

```diff
@@ -45,30 +45,28 @@ Each request type will defines its own `requests` object using with its own
 
 #### Block Header
 
-Extend the header with a new 32 byte value `requests_hash`.
+Extend the header with a new 32 byte commitment value `requests_hash`.
 
-The construction looks like:
+A requests hash list is computed by hashing the elements of the block requests list, but
+only ones where `request_data` is non-empty. The commitment is then computed as the sha256
+hash of the list of requests element hashes.
 
-```
-sha256(sha256(requests_0) ++ sha256(requests_1) ++ ...)
-```
-
-Or in pseudocode:
 
 ```python
-def compute_requests_hash(requests):
+def compute_requests_hash(block_requests):
     m = sha256()
-    for r in requests:
-        m.update(sha256(r))
+    for r in block_requests:
+        if len(r) > 1:
```

Comment:
Ok, this works, but this assumes that `request_type` is one byte, which is not specified.

I would also somewhere note that this thus checks that there is any `request_data` (without `request_data` then `r` len is indeed 1 (assuming that it is a `Bytes1` type)). I initially thought this was off and should be `len(r) >= 1`.

Reply:
The type will always be one byte. We should probably add that to the EIP.